test/integration/cluster/cluster_test.go

```diff
@@ -0,0 +1,139 @@
+/*
+Copyright 2019 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package integration
+
+import (
+	"os"
+	"testing"
+	"time"
+
+	. "github.com/onsi/ginkgo"
+	. "github.com/onsi/gomega"
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	informers "k8s.io/client-go/informers"
+	"k8s.io/client-go/kubernetes"
+	"k8s.io/client-go/tools/cache"
+	"k8s.io/client-go/tools/clientcmd"
+	clusterv1alpha1 "sigs.k8s.io/cluster-api/pkg/apis/cluster/v1alpha1"
+	clientset "sigs.k8s.io/cluster-api/pkg/client/clientset_generated/clientset"
+	client "sigs.k8s.io/cluster-api/pkg/client/clientset_generated/clientset/typed/cluster/v1alpha1"
+)
+
+var clusterSpec = &clusterv1alpha1.ClusterSpec{
+	ClusterNetwork: clusterv1alpha1.ClusterNetworkingConfig{
+		ServiceDomain: "mydomain.com",
+		Services: clusterv1alpha1.NetworkRanges{
+			CIDRBlocks: []string{"10.96.0.0/12"},
+		},
+		Pods: clusterv1alpha1.NetworkRanges{
+			CIDRBlocks: []string{"192.168.0.0/16"},
+		},
+	},
+}
+
+// Constructs a function for async event retrieval
+// amenable to be used with Gomega's Eventually assertion
+func checkEvents(c chan *corev1.Event) func() *corev1.Event {
+	return func() *corev1.Event {
+		select {
+		case e := <-c:
+			return e
+		default:
+			return nil
+		}
+	}
+}
+
+const TIMEOUT = 10 * time.Second
```

Comment:
not sure about a perfect reasonable timeout value, 10s is not enough in my local testing. 1m should be good.
i still think it is good to have a timeout on this call even the test has a overall timeout

Reply:
@figo This is the time for the event to come after the clusterapi operation is executed (e.g. create a cluster). Not for the whole test to run.
That said,  would it work if I add an `async-timeout` flag to the test so different timeouts can be defined?  In order to do so, I must add a `TestMain` function and parse the parameter:

```
var asyncTimeout Duration

func TestMain(m *testing.M) { 
   duration := flag.DurationVar(&asyncTimeout, 10*time.Second, "async-timeout", "Asynchronous test timeout")
   flag.Parse() 

    retCode := m.Run()

    os.Exit(retCode)
}
```

Reply:
@pablochacin i am suggesting to increase the timeout value, i understand it should not taking long from a operation to event been captured, but testing locally shows delay can happen (could be due to slow cluster).  the purpose of setting timeout probably is: event should be detected in a reasonable time, otherwise something went wrong in the cluster.  would 1minute been a reasonable time? or defer to overall test timeout (10 minutes?)

Reply:
Ok with one minute